src/dlangui/graphics/colors.d

```diff
@@ -191,20 +190,26 @@ uint makeRGBA(T)(T r, T g, T b, T a) pure nothrow {
 
 /// blend two RGB pixels using alpha
 uint blendARGB(uint dst, uint src, uint alpha) pure nothrow {
-    uint dstalpha = dst >> 24;
-    if (dstalpha > 0x80)
-        return src;
+    uint srca = (src >> 24) & 0xFF;
```

Comment:
Don't `>> 24` and `& 0xFF` achieve the same result? Please, explain why you added this

Reply:
Here it was done so that if the first color is more than half transparent, then mixing does not occur, but the second color is returned. I don't know why. I made it so that each color has an alpha channel selected and then blended.

Reply:
What I mean was, the `& 0xFF` part is unnecessary, as `>> 24` already leaves you with the highest order byte:
```d
import std;

void main() @safe
{
	auto a = 0xAABBCCDD;
    writeln(toChars!(16)(a >> 24)); // Prints out `aa`
    writeln(toChars!(16)( (a >> 24) & 0xFF)); // Also prints out `aa`
}
```

Reply:
You're right, since here the uint number is shifted 24 bits to the right, the remaining 8 bits will always be in the correct range. You can think it can be removed for such cases, but this is done for the uniformity of the code.